app/src/main/java/com/rob729/quiethours/Fragments/MainFragment.kt

```diff
@@ -105,6 +109,36 @@ class MainFragment : Fragment() {
                     .navigate(MainFragmentDirections.actionMainFragmentToNewProfileFragment())
             }
         }
+        val sharedPreferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
```

Comment:
Lazily initialize the sharedPreference outside the onCreate().
Also, shift this whole part of the code added in this PR in a function and call that function in onCreate.